src/components/Main/Main.jsx

```diff
@@ -0,0 +1,257 @@
+//       json.data.shift()
+//       console.log("json.data", json.data)
+//       setRows(json.data)
+//     })
+// }, [])
+
+const Main = () => {
+
+    const [rows, setRows] = useState([]);
+    const [sortType, setSortType] = useState();
+
+    useEffect(() => {
+        fetch('./data.csv')
+            .then(res => res.text())
+            .then(csv => {
+                const json = Papaparse.parse(csv)
+                setRows(transformData(json.data))
+                copyOfRows = transformData(json.data)
+            })
+    }, [])
+
+    const transformData = (payload) => {
+        const headers = payload[0];
+        const [first, ...data] = payload;
+
+        const transformedData = data.map((item, index) => {
+            let obj = {};
+            item.forEach((i, idx) => {
+                obj[headers[idx]] = i;
+            })
+            return obj;
+        });
+
+        return transformedData;
+    }
+
+    const sortTable = useMemo(() => {
+        let result = [...rows];
+
+        if (sortType === "dateUp") {
+            result.sort((a, b) => new Date(b["Date"]) - new Date(a["Date"]));
+        }
+        else if (sortType === "dateDown") {
+            result.sort((a, b) => new Date(a["Date"]) - new Date(b["Date"]));
+        }
+        else if (sortType === "companyUp") {
+            result.sort((a, b) => a["Company"] > b["Company"] ? 1 : -1);
+        }
+        else if (sortType === "companyDown") {
+            result.sort((a, b) => b["Company"] > a["Company"] ? 1 : -1);
+        }
+        else if (sortType === "positionUp") {
+            result.sort((a, b) => a["Position"] > b["Position"] ? 1 : -1);
+        }
+        else if (sortType === "positionDown") {
+            result.sort((a, b) => b["Position"] > a["Position"] ? 1 : -1);
+        }
+        else if (sortType === "salaryUp") {
+            result.sort((a, b) => {
+                if (a["Salary"] !== undefined && b["Salary"] !== undefined) {
+                    return +a["Salary"].split(',').join('') > +b["Salary"].split(',').join('') ? 1 : -1
+                }
+            });
+        }
+        else if (sortType === "salaryDown") {
+            result.sort((a, b) => {
+                if (a["Salary"] !== undefined && b["Salary"] !== undefined) {
+                    return +b["Salary"].split(',').join('') > +a["Salary"].split(',').join('') ? 1 : -1
+                }
+            });
+        }
+        else if (sortType === "expUp") {
+            result.sort((a, b) => {
+                if (a["Years of experience"] !== undefined && b["Years of experience"] !== undefined) {
+                    return +a["Years of experience"].split(',').join('') > +b["Years of experience"].split(',').join('') ? 1 : -1
+                }
+            });
+        }
+        else if (sortType === "expDown") {
+            result.sort((a, b) => {
+                if (a["Years of experience"] !== undefined && b["Years of experience"] !== undefined) {
```

Comment:
u can change this line by if (a["Years of experience"] && b["Years of experience"])

Reply:
Also please use variables for this